logical/framework/field_data.go

```diff
@@ -291,6 +294,103 @@ func (d *FieldData) getPrimitive(k string, schema *FieldSchema) (interface{}, bo
 		}
 		return result, true, nil
 
+	case TypeHeader:
+		/*
+
+			There are multiple ways a header could be provided:
+
+			1.	As a map[string]interface{} that resolves to a map[string]string or map[string][]string, or a mix of both
+				because that's permitted for headers.
+				This mainly comes from the API.
+
+			2.	As a string...
+				a. That contains JSON that originally was JSON, but then was base64 encoded.
+				b. That contains JSON, ex. `{"content-type":"text/json","accept":["encoding/json"]}`.
+				This mainly comes from the API and is used to save space while sending in the header.
+
+			3.	As an array of strings that contains comma-delimited key-value pairs associated via a colon,
+				ex: `content-type:text/json`,`accept:encoding/json`.
+				This mainly comes from the CLI.
+
+			We go through these sequentially below.
+
+		*/
+		result := http.Header{}
+
+		toHeader := func(resultMap map[string]interface{}) (http.Header, error) {
+			header := http.Header{}
+			for headerKey, headerValGroup := range resultMap {
+				switch typedHeader := headerValGroup.(type) {
+				case string:
+					header.Add(headerKey, typedHeader)
+				case []string:
+					for _, headerVal := range typedHeader {
```

Comment:
Doesn't this need to be `range typedHeader.([]string)`? Same with the case below.

Reply:
Yes and no. It's already converted to an []string here: `switch typedHeader := headerValGroup.(type)`. The type is used in the switch below and it's actually cast to it when it's instantiated there on the left. Went off of [this](https://tour.golang.org/methods/16).

Reply:
Ah right, I mixed up:
```
switch v.(type) {
```
which doesn't have that type with what you did here, which does. Great!